### PR TITLE
Enable internal JWT by default

### PR DESCRIPTION
**Pull request by dain**

### CODE REVIEW

File: presto-main/src/main/java/io/prestosql/server/InternalAuthenticationManager.java [hunk 1]
```diff
@@ -49,16 +48,21 @@
     @Inject
     public InternalAuthenticationManager(InternalCommunicationConfig internalCommunicationConfig, NodeInfo nodeInfo)
     {
-        this(requireNonNull(internalCommunicationConfig, "internalCommunicationConfig is null").getSharedSecret(), nodeInfo.getNodeId());
+        this(
+                requireNonNull(internalCommunicationConfig, "internalCommunicationConfig is null")
+                        .getSharedSecret()
+                        .orElse(requireNonNull(nodeInfo, "nodeInfo is null").getEnvironment()),
```
Comment (findepi) [inline presto-main/src/main/java/io/prestosql/server/InternalAuthenticationManager.java]: is environment a good shared secret replacement?

File: presto-main/src/main/java/io/prestosql/server/InternalAuthenticationManager.java [hunk 2]
```diff
@@ -68,13 +72,13 @@ public InternalAuthenticationManager(Optional<String> sharedSecret, String nodeI
 
     public boolean isInternalRequest(HttpServletRequest request)
     {
-        return request.getHeader(PRESTO_INTERNAL_BEARER) != null;
+        return internalJwtEnabled && request.getHeader(PRESTO_INTERNAL_BEARER) != null;
```
Comment (sopel39) [inline presto-main/src/main/java/io/prestosql/server/InternalAuthenticationManager.java]: is this required. When `internalJwtEnabled` is `false` then `filterRequest` won't add `PRESTO_INTERNAL_BEARER`

File: presto-main/src/main/java/io/prestosql/server/InternalAuthenticationManager.java (hunk 1 above)
Comment (dain) [reply presto-main/src/main/java/io/prestosql/server/InternalAuthenticationManager.java]: It is shared, but not a secret.  This works, but is not secure.  If you want to secure internal communication, you must set the secret.

File: presto-main/src/main/java/io/prestosql/server/InternalAuthenticationManager.java (hunk 2 above)
Comment (dain) [reply presto-main/src/main/java/io/prestosql/server/InternalAuthenticationManager.java]: This is not needed.

File: presto-main/src/main/java/io/prestosql/server/InternalAuthenticationManager.java (hunk 1 above)
Comment (findepi) [reply presto-main/src/main/java/io/prestosql/server/InternalAuthenticationManager.java]: > If you want to secure internal communication, you must set the secret.

this i agree with.

question is -- can we have this _as a default_?

Today, when someone secures a cluster they often do this:

1. configure certs, enable HTTPS, disable HTTP
2. enable PASSWORD authenticator, usually with LDAP
3. they _realize_ workers cannot connect, so they figure they need to enable eg certs or JWT for internal comms
4. they add certs or JWT
5. everything works and they have a secure cluster

after this change, the process may look like this:

1. configure certs, enable HTTPS, disable HTTP
2. enable PASSWORD authenticator, usually with LDAP
5. everything works and they _think_ they have a secure cluster

I strongly feel this is "insecure by default" in an unobvious way.

File: presto-main/src/main/java/io/prestosql/server/InternalAuthenticationManager.java [hunk 3]
```diff
@@ -74,7 +78,7 @@ public boolean isInternalRequest(HttpServletRequest request)
     public Principal authenticateInternalRequest(HttpServletRequest request)
     {
         if (!internalJwtEnabled) {
-            log.error("Internal authentication in not configured");
+            log.error("Internal authentication in not enabled");
```
Comment (electrum) [inline presto-main/src/main/java/io/prestosql/server/InternalAuthenticationManager.java]: Typo: in -> is

File: presto-main/src/main/java/io/prestosql/server/InternalCommunicationConfig.java [hunk 4]
```diff
@@ -47,6 +48,18 @@ public InternalCommunicationConfig setSharedSecret(String sharedSecret)
         return this;
     }
 
+    public boolean isInternalJwtEnabled()
+    {
+        return internalJwtEnabled;
+    }
+
+    @Config("internal-communication.jwt.enabled")
```
Comment (electrum) [inline presto-main/src/main/java/io/prestosql/server/InternalCommunicationConfig.java]: Should this be marked as legacy/deprecated? Is there a use case to turn this off? Maybe for debugging? If so, perhaps name it something like `internal-communication.jwt.disabled-for-testing`

(we can merge this as-is for now and follow up later)

**Pull request merged**